### ISSUE TITLE
Vale spell check affected by surrounding lines

### ISSUE THREAD

**Issue by vtlim** [jdkato/txtlint#455]
Vale's spell check does not always capture misspellings depending on the surrounding lines. For the following snippet,
```
This is a typpo.
A sentence with [remark-gfm](i-am-a-link).
A sentence with remark-gfm.
A sentence with remark-gfm.
A sentence with gfm.
A sentence with remark.
A sentence with remmark.

A sentence with [remark-gfm](i-am-a-link).
```
We've noticed the following:
* Line 3 does not trigger a spelling error on `gfm` unless line 2 is removed and line 3 moves up to line 2.
* Line 5 does not trigger a spelling error unless a new line is added before it.
* However, putting a new line before line 5 causes line 2 to stop triggering an error.

Screen capture of the findings:
https://user-images.githubusercontent.com/7747997/178799223-483de64f-e432-4302-b1f5-49203534032a.mp4

A few screenshots to try to capture the issue:
* Lines 3, 4, 5, 9 should have an error
![image](https://user-images.githubusercontent.com/7747997/178802477-1fb80bda-e3dd-45cf-be31-37f9d49ca0ba.png)
* Lines 3, 4, 8 should have an error
![image](https://user-images.githubusercontent.com/7747997/178802821-7b0b2bcb-c84c-46bf-9a48-07ac8d97d65c.png)
* Lines 2, 3, 4, 10 should have an error
![image](https://user-images.githubusercontent.com/7747997/178802657-4d78ccbe-8ab0-4b52-87a0-0ddfc8aeeaaf.png)

We installed vale using Homebrew on macOS. We saw these results in VS code as well as with the Vale CLI.
Tested with vale versions **2.15.5** and **2.18.0**

Vale config file, `.vale.ini`:
```
StylesPath = "valestyles"
MinAlertLevel = suggestion
Vale.Terms = NO
[formats]
[*.md]
BasedOnStyles = Vale
```

**Comment by jdkato** [comment]
Thanks for the detailed report.

Your last screenshot actually shows the expected result: with the default configuration, only "typpo", "gfm", and "remmark" should be reported as errors.

The bug here is that the location of "gfm" is being misreported in some cases, which should be fixed in the next release.

**Comment by vtlim** [comment]
Thanks @jdkato! Can you clarify what you mean by the default configuration? Is there something we would need to configure to spell check each part of a hyphenated word? Related issue: https://github.com/errata-ai/vale/issues/199
